lib/1.4/utility.dml

```diff
@@ -342,11 +343,15 @@ template _init_val_hard_reset is (init_val, hard_reset) {
 // Instantiate this on top level to support hard reset.
 template hreset is hard_reset {
     port HRESET {
+        saved bool reset_asserted;
         implement signal {
             method signal_raise() default {
+                reset_asserted = true;
                 dev.hard_reset();
             }
             method signal_lower() default {
+                dev.hard_reset_deassert();
+                reset_asserted = false;
             }
```

Comment:
It would make more sense to start with `template hreset is (hard_reset, signal_port)` and continue from there. IIRC, @svenwest tried this when adding `signal_port`, but it was dropped since it caused too much short-term SSM breakage. I think it can be done, but requires a careful migration plan.

Reply:
Both templates are define default methods. Using spec violation is also doubtful.

Reply:
Is it possible to call default() from the default method?
            method signal_lower() default {
                default();
            }

Reply:
I would prefer to copy a logic from signal port template
`template hreset is hard_reset {
    port HRESET {
        saved bool high;
        implement signal {
            method signal_raise() default {
                high = true;
                dev.hard_reset();
            }
            method signal_lower() default {
                high = false;
            }`